pkgs/applications/video/handbrake/default.nix

```diff
@@ -43,12 +45,13 @@ stdenv.mkDerivation rec {
     fribidi fontconfig freetype jansson zlib
     libass libiconv libsamplerate libxml2 bzip2
     libogg libopus libtheora libvorbis libdvdcss a52dec libmkv
-    lame ffmpeg libdvdread libdvdnav libbluray mp4v2 mpeg2dec x264 x265 libvpx
-  ] ++ (lib.optionals useGtk [
+    lame ffmpeg libdvdread libdvdnav libbluray mp4v2 mpeg2dec x264 x265 libvpx]
+    ++ lib.optionals useGtk [
     glib gtk3 libappindicator-gtk3 libnotify
     gst_all_1.gstreamer gst_all_1.gst-plugins-base dbus-glib udev
-    libgudev
-  ]) ++ (lib.optionals useFdk [fdk_aac]);
+    libgudev]
+    ++ (if useFfmpeg then [ffmpeg] else [libav_12])
```

Comment:
I made it this way. Is it right way?

Reply:
if handbrake is not supposed to work with ffmpeg, there is no point in this flag.

Reply:
I followed the history of HandBrake through the years. And also FFMpeg -> LibAV fork and their competition and feuds. So I understand their stands, who main developers of HandBrake, and why they use LibAV.

People can use HandBrake with FFMpeg if they would have need, or gains from that. But HB upstream would say to them - "is not supported".

Many people do compile and use it it with FFMpeg.
And developers somehow support FFMpeg (defacto).

They stance is much because of feuds they had in the past. And their experience with distribution package maintainers.

Like, current NixOS version works only on FFMpeg, and I do not see people changing that through the years.

Since we do it as nondefault option - it is only an additional option.

If we serious about that, HandBrake distributes with it's own versions of LibAV and all other libraries. We successfully use upstream NixOS packages - but that is not supported by HB upstream.

Reply:
The main thing is to make default way - the upstream supported. While allow people the freedom also.